server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -490,6 +501,16 @@ public void addIgnoredField(String field) {
         public Collection<String> getIgnoredFields() {
             return Collections.unmodifiableCollection(ignoredFields);
         }
+
+        @Override
+        public void addFieldExistsField(String field) {
```

Comment:
nit: I slightly prefer the previous name, shall we call it addFieldToFieldNames ?

Reply:
I ended up with `addtoFieldNames`